runtime/sema/variable_activations.go

```diff
@@ -156,22 +156,21 @@ func (a *VariableActivations) DeclareImplicitConstant(
 	)
 }
 
-func (a *VariableActivations) VariablesDeclaredInAndBelow(depth int) map[string]*Variable {
-	variables := map[string]*Variable{}
+func (a *VariableActivations) ForEachVariablesDeclaredInAndBelow(depth int, f func(name string, value *Variable)) {
```

Comment:
Oh! I like this... Iterate in place instead of building a list and iterating over the list. 👍